qiskit/algorithms/eigen_solvers/vqd.py

```diff
@@ -0,0 +1,840 @@
+                    "deprecated as of Qiskit Terra 0.19.0 and will be unsupported no "
+                    "sooner than 3 months after the release date. Instead use an optimizer "
+                    "providing ``minimize`` (see qiskit.algorithms.optimizers.Optimizer).",
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
+
+                opt_result = self.optimizer.optimize(
+                    len(initial_point),
+                    energy_evaluation,
+                    gradient if step == 1 else None,
+                    bounds,
+                    initial_point,
+                )
+
+            eval_time = time() - start_time
+
+            result.optimal_point.append(opt_result.x)
+            result.optimal_parameters.append(dict(zip(self._ansatz_params, opt_result.x)))
+            result.optimal_value.append(opt_result.fun)
+            result.cost_function_evals.append(opt_result.nfev)
+            result.optimizer_time.append(eval_time)
+
+            eigenvalue = (
+                StateFn(operator, is_measurement=True)
+                .compose(CircuitStateFn(self.ansatz.bind_parameters(result.optimal_parameters[-1])))
+                .reduce()
+                .eval()
+            )
+
+            result.eigenvalues.append(eigenvalue)
+            result.eigenstates.append(self._get_eigenstate(result.optimal_parameters[-1]))
+
+            if aux_operators is not None:
+                bound_ansatz = self.ansatz.bind_parameters(result.optimal_point)
+                aux_value = eval_observables(
+                    self.quantum_instance, bound_ansatz, aux_operators, expectation=expectation
+                )
+                aux_values.append(aux_value)
+
+            if step == 1:
+
+                logger.info(
+                    "Ground state optimization complete in %s seconds.\nFound opt_params %s in %s evals",
+                    eval_time,
+                    result.optimal_point,
+                    self._eval_count,
+                )
+            else:
+                logger.info(
+                    (
+                        "%s excited state optimization complete in %s s.\nFound opt_parms %s in %s evals"
+                    ),
+                    str(step - 1),
+                    eval_time,
+                    result.optimal_point,
+                    self._eval_count,
+                )
+
+        # To match the siignature of NumpyEigenSolver Result
+        result.eigenstates = ListOp([StateFn(vec) for vec in result.eigenstates])
+        result.eigenvalues = np.array(result.eigenvalues)
+        result.optimal_point = np.array(result.optimal_point)
+        result.optimal_value = np.array(result.optimal_value)
+        result.cost_function_evals = np.array(result.cost_function_evals)
+        result.optimizer_time = np.array(result.optimizer_time)
+
+        # TODO delete as soon as get_optimal_vector etc are removed
+        self._ret = result
```

Comment:
This can be deleted too - similar to another comment before on similar code above